### PR TITLE
PROP-5 : System should show autocomplete search box on landing page.

### PR DESCRIPTION
**Pull request by sakshigokhru**
## Description
_System should show autocomplete search box on landing page._
_Setup API call which return array of property address matching from the query string._

### CODE REVIEW

File: src/controllers/property.js [hunk 1]
```diff
@@ -13,3 +15,19 @@ exports.storePropertyAddress = (req, res, next) => {
     }
   });
 };
+
+exports.getPropertyAddress = function (req, res) {
```
Comment (bbbneo333) [inline src/controllers/property.js]: Please follow arrow `=>` style notation when defining controllers.

File: src/controllers/property.js [hunk 2]
```diff
@@ -13,3 +15,19 @@ exports.storePropertyAddress = (req, res, next) => {
...
+    properties.forEach(function (item) {
+      if (item.propertyAddress && item.propertyAddress.match(regex)) {
+        propertyAddressList.push(item.propertyAddress);
+      }
+    });
+    const uniqueProperties = _.uniq(propertyAddressList);
+    uniqueProperties.sort();
+    return res.status(200).json(uniqueProperties);
+  })
+    .catch(() => res.status(404).send('No properties matched'));
```
Comment (jjalan) [inline src/controllers/property.js]: @bbbneo333 - Don't we have a mechanism to throw an error and a global handler to catch and return to the client?

File: package.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
     "i18n": "^0.8.3",
     "mongoose": "^5.2.2",
     "raven": "^2.6.3",
+    "underscore": "^1.8.3",
```
Comment (jjalan) [inline package.json]: We should use lodash instead of underscore.

File: src/controllers/property.js (hunk 2 above)
Comment (bbbneo333) [reply src/controllers/property.js]: Yes we do have it. Please use `next(Error.NotFound('your-message-here'))`. `Error` can be found at `helpers`.

**Comment by bbbneo333** [comment]
Please add `MONGODB_URI` to `app.json` as well and set it to `required: true` as discussed.

**Pull request merged**